src/core/algorithms/dc/model/column_operand.h

```diff
@@ -3,79 +3,136 @@
 #include <cstdlib>
 #include <exception>
 #include <memory>
+#include <optional>
 #include <string>
 #include <vector>
 
-#include <boost/functional/hash.hpp>
 #include <easylogging++.h>
 
 #include "model/table/column.h"
 #include "model/table/relation_data.h"
+#include "model/types/type.h"
 
 namespace algos::dc {
 
+enum class Tuple { kS, kT, kMixed };
+
 //  @brief Represents a column operand for Predicate.
 //
 //  A predicate (e.g., t.A == s.A) comprises three elements:
 //  the column operand from the first tuple ("t.A"), the comparison operator
 //  ("=="), and the column operand from the second tuple ("s.A"). The `ColumnOperand` class
 //  encapsulates the column operand part of a predicate, such as "t.A" or "s.A".
 //
+//  A constant value also can be a column operand thus boost::optional is utilized
+//
 //  The class distinguishes between operands derived from the first tuple (t) and those
 //  from the second tuple (s) using a boolean flag `is_first_tuple_`, where `true` indicates an
 //  operand from the first tuple (t), and `false` indicates an operand from the second
 //  tuple (s).
 class ColumnOperand {
 private:
     Column const* column_;
-    bool is_first_tuple_;
+    std::optional<dc::Tuple> tuple_;
+    model::Type const* type_;
+    std::byte const* val_;
 
 public:
-    ColumnOperand(Column const* column, bool is_first_tuple) noexcept
-        : column_(column), is_first_tuple_(is_first_tuple) {}
-
-    ColumnOperand() noexcept = default;
-
-    // For conversion from "t.ColumnPosition" or "t.ColumnName"
-    ColumnOperand(std::string operand, RelationalSchema const& schema) {
-        if (operand.front() != 't' and operand.front() != 's')
-            throw std::logic_error("Unknown tuple name");
-
-        is_first_tuple_ = operand.front() == 't';
-        std::string name(operand.begin() + 2, operand.end());
-        std::vector<std::unique_ptr<Column>> const& cols = schema.GetColumns();
-        if (!cols.front()->GetName().empty()) {  // Has header
-            for (std::unique_ptr<Column> const& col : cols) {
-                if (name == col->GetName()) {
-                    column_ = col.get();
-                    return;
-                }
-            }
+    ColumnOperand() noexcept : column_(nullptr), val_(nullptr) {};
+
+    ColumnOperand(Column const* column, dc::Tuple tuple, model::Type const* type)
+        : column_(column), tuple_(tuple), type_(type), val_(nullptr) {}
+
+    ColumnOperand(std::string const& str_val, model::Type const* type)
+        : column_(nullptr), type_(type) {
+        std::byte* val = type->Allocate();
+        type->ValueFromStr(val, str_val);
```

Comment:
Feel free to ignore:
I would use `type_` here. Makes more sense to me that in constructor we filled some private fields first and then called methods on _those_ fields, instead on what we filled them with (though it's the same pointer, naturally)